src/opower/utilities/apco.py

```diff
@@ -0,0 +1,133 @@
+"""Appalachian Power Company (APCO)."""
+
+from html.parser import HTMLParser
+import re
+
+import aiohttp
+
+from ..const import USER_AGENT
+from ..exceptions import InvalidAuth
+from .base import UtilityBase
+
+
+class APCOLoginParser(HTMLParser):
+    """HTML parser to extract login verification token from APCO Login page."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self.verification_token = None
+
+    def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
+        """Try to extract the verification token from the login input."""
+        if tag == "input" and ("name", "__RequestVerificationToken") in attrs:
+            _, token = next(filter(lambda attr: attr[0] == "value", attrs))
+            self.verification_token = token
+
+
+class APCOUsageParser(HTMLParser):
+    """HTML parser to extract OPower bearer token from APCO Usage page."""
+
+    _regexp = re.compile(r'var accessToken\s+=\s+["\'](?P<token>.+)["\']')
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self.opower_access_token = None
+        self._in_inline_script = False
+
+    def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
+        """Recognizes inline scripts."""
+        if (
+            tag == "script"
+            and next(filter(lambda attr: attr[0] == "src", attrs), None) is None
+        ):
+            self._in_inline_script = True
+
+    def handle_data(self, data: str) -> None:
+        """Try to extract the access token from the inline script."""
+        if self._in_inline_script:
+            result = self._regexp.search(data)
+            if result and result.group("token"):
+                self.opower_access_token = result.group("token")
+
+    def handle_endtag(self, tag: str) -> None:
+        """Recognizes the end of inline scripts."""
+        if tag == "script":
+            self._in_inline_script = False
+
+
+class APCO(UtilityBase):
```

Comment:
Rename to AppalachianPower. Also rename the file to appalachianpower.py. The convention is to match the utility domain.